src/pyk/kore_exec_covr/__main__.py

```diff
@@ -0,0 +1,165 @@
+    """
+    Inputs:
+       * definition compiled with "kompile --backend haskell --emit-json"
+       * Log file produced with "KORE_EXEC_OPTS='--log-format oneline --log-entries DebugAppliedRewriteRules,DebugApplyEquation'"
+
+    Outputs:
+       * human-readable report of applied rewrite and simplification rules,
+         with labels (if declared) and locations
+    """
+    definition = read_kast_definition(definition_dir / 'compiled.json')
+
+    _LOGGER.info(f'Discovering rewrite and simplification rules in {definition_dir}')
+    rule_dict = build_rule_dict(definition)
+
+    rewrites: Dict[str, int] = {key: 0 for key, rule in rule_dict.items() if not 'simplification' in rule.att.atts}
+    simplifications: Dict[str, int] = {key: 0 for key, rule in rule_dict.items() if 'simplification' in rule.att.atts}
+    _LOGGER.info(f'    Found {len(rewrites)} rewrite rules')
+    _LOGGER.info(f'    Found {len(simplifications)} simplification rules')
+
+    log_entries = input_file.read_text().splitlines()
+    for log_entry in log_entries:
+        entry_type, location_str = parse_haskell_one_line_log(log_entry)
+        # TODO: the haskell backend log files often contain items with no location. it is not clear to me what happens.
+        if location_str:
+            try:
+                if entry_type == HaskellLogEntry.DebugAppliedRewriteRules:
+                    rewrites[location_str] += 1
+                elif entry_type == HaskellLogEntry.DebugApplyEquation:
+                    simplifications[location_str] += 1
+            except KeyError:
+                _LOGGER.warning(f'unknown rule location: {location_str}')
+
+    print('=================================')
+    print('=== REWRITES ====================')
+    print('=================================')
+    for location_str, hits in rewrites.items():
+        rule_label_str = ''
+        if 'label' in rule_dict[location_str].att.atts:
+            rule_label_str = f'[{rule_dict[location_str].att.atts["label"]}]'
+        if hits > 0:
+            print(f'    {hits} applications of rule {rule_label_str} defined at {location_str}')
+    total_rewrites = sum([v for v in rewrites.values() if v > 0])
+    print(f'Total rewrites: {total_rewrites}')
+
+    print('=================================')
+    print('=== SIMPLIFICATIONS =============')
+    print('=================================')
+    for location_str, hits in simplifications.items():
+        rule_label_str = ''
+        if 'label' in rule_dict[location_str].att.atts:
+            rule_label_str = f'[{rule_dict[location_str].att.atts["label"]}]'
+        if hits > 0:
+            print(f'    {hits} applications of rule {rule_label_str} defined at {location_str}')
+    total_simplifications = sum([v for v in simplifications.values() if v > 0])
+    print(f'Total simplifications: {total_simplifications}')
+
+
+def location_tuple_to_str(location: Tuple[int, int, int, int]) -> str:
```

Comment:
For every function not intended to be exposed from the module:

```suggestion
def _location_tuple_to_str(location: Tuple[int, int, int, int]) -> str:
```